### ISSUE TITLE
SSL Configuration Fails Unless Fully Configured

### ISSUE THREAD

**Issue by netclectic** [akkadotnet/akka.net#3594]
When setting up SSL configuration I have to set valid path, password and flags properties even when enable-ssl is set to false.

- Which Akka.Net version you are using
  * 1.3.9

- On which platform you are using Akka.Net
  * .Net Framework 4.7.2 on Windows 10 x64

- A list of steps to reproduce the issue. Or an gist or github repo which can be easily used to reproduce your case.

  * a node configured with the following hocon fails:
```
	enable-ssl = false
	
	ssl 
	{
		{
			path = "<valid certificate path>" 
			password = "<certificate password>"
			
			flags = [ "default-key-set" ] 
		}
	}                        
```

I would expect to be able to provide 'invalid' sample ssl configuration while 'enable-ssl' is set to false, the ssl configuration details should not be validated unless 'enable-ssl' is set to true.

N.B. the documentation says the default flags should be: "default-flag-set", while it appears the actual valid value is: "default-key-set".

**Comment by Aaronontheweb** [comment]
> I would expect to be able to provide 'invalid' sample ssl configuration while 'enable-ssl' is set to false, the ssl configuration details should not be validated unless 'enable-ssl' is set to true.

yes, that sounds right to me - can we setup a reproduction spec for this? Because if what you're saying is true then Akka.Remote should always be catching fire whenever anyone uses it without explicitly setting up SSL, no?

**Comment by netclectic** [comment]
> yes, that sounds right to me - can we setup a reproduction spec for this? Because if what you're saying is true then Akka.Remote should always be catching fire whenever anyone uses it without explicitly setting up SSL, no?

No, only if they've explicitly setup **invalid** SSL but regardless of whether 'enable-ssl' is true or false.

I haven't revisited this since I first reported it, I'll have a look and see if I can produce a spec.

**Comment by aminchenkov** [comment]
Hello. Any update on it? It is an issue for us as well.

**Comment by aminchenkov** [comment]
@Aaronontheweb , could you review https://github.com/akkadotnet/akka.net/pull/6038?
I decided that EnableSsl should be primary setting, regardless if SSL section is defined or not. In our project it always will be setup, but EnableSsl configuration might be set to false, depending on an environment.

**Comment by Aaronontheweb** [comment]
That PR looks good. We'll include this fix inside the 1.4.40 release.

**Comment by aminchenkov** [comment]
> That PR looks good. We'll include this fix inside the 1.4.40 release.

@Aaronontheweb , do you know release date?

**Comment by Aaronontheweb** [comment]
I don't - https://github.com/akkadotnet/akka.net/pull/6043 must be merged first since `dev` isn't the right branch for stable releases. `v1.4` is. @eaba is working on it.

**Comment by aminchenkov** [comment]
@Aaronontheweb , thank you!

**Comment by Aaronontheweb** [comment]
Fixed in 1.4.40.
